presto-main/src/test/java/com/facebook/presto/sql/planner/planPrinter/TestJsonRenderer.java

```diff
@@ -0,0 +1,163 @@
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.SessionTestUtils.TEST_SESSION;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static com.facebook.presto.metadata.AbstractMockMetadata.dummyMetadata;
+import static com.facebook.presto.metadata.FunctionAndTypeManager.createTestFunctionAndTypeManager;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static org.testng.Assert.assertEquals;
+
+public class TestJsonRenderer
+{
+    private static final FunctionAndTypeManager functionAndTypeManager = createTestFunctionAndTypeManager();
+    private static final PlanBuilder PLAN_BUILDER = new PlanBuilder(TEST_SESSION, new PlanNodeIdAllocator(), dummyMetadata());
+    private static final VariableReferenceExpression COLUMN_VARIABLE = new VariableReferenceExpression(Optional.empty(), "column", VARCHAR);
+    private static final ColumnHandle COLUMN_HANDLE = new TestingMetadata.TestingColumnHandle("column");
+    private static final PlanVariableAllocator variableAllocator = new PlanVariableAllocator();
+    private static final JsonRenderer jsonRenderer = new JsonRenderer(functionAndTypeManager);
+    private static final JsonObjectMapperProvider provider = new JsonObjectMapperProvider();
+    private static final TableHandle TABLE_HANDLE_WITH_LAYOUT = new TableHandle(
+            new ConnectorId("testConnector"),
+            new TestingMetadata.TestingTableHandle(),
+            TestingTransactionHandle.create(),
+            Optional.of(TestingHandle.INSTANCE));
+    private static JsonCodec<JsonRenderer.JsonRenderedNode> planCodec;
+
+    static {
+        provider.setKeyDeserializers(ImmutableMap.of(VariableReferenceExpression.class, new Serialization.VariableReferenceExpressionDeserializer(createTestFunctionAndTypeManager())));
+        JsonCodecFactory codecFactory = new JsonCodecFactory(provider, true);
+        planCodec = codecFactory.jsonCodec(JsonRenderer.JsonRenderedNode.class);
+    }
+    private PlanRepresentation getPlanRepresentation(PlanNode root)
+    {
+        return new PlanRepresentation(
+                root,
+                variableAllocator.getTypes(),
+                Optional.empty(),
+                Optional.empty());
+    }
+    private NodeRepresentation getNodeRepresentation(PlanNode root, List<PlanNodeId> planNodeIds)
+    {
+        return new NodeRepresentation(
+                Optional.empty(),
+                root.getId(),
+                root.getClass().getName(),
+                root.getClass().getSimpleName(),
+                "",
+                root.getOutputVariables(),
+                Optional.empty(),
+                ImmutableList.of(),
+                ImmutableList.of(),
+                planNodeIds,
+                ImmutableList.of());
+    }
```

Comment:
nit: add empty line below

Reply:
All functions shall have an empty line below. You can use "Code -> Reformat Code" in IntelliJ to format the whole file, or all files in a folder